sdk/ai/ai-inference-rest/src/trace.ts

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  PathUncheckedResponse,
+  RequestParameters,
+  StreamableMethod,
+} from "@azure-rest/core-client";
+import {
+  createTracingClient,
+  OperationTracingOptions,
+  TracingSpan,
+  TracingSpanKind,
+} from "@azure/core-tracing";
+import {
+  GetChatCompletionsBodyParam,
+  GetEmbeddingsBodyParam,
+  GetImageEmbeddingsBodyParam,
+} from "./parameters.js";
+import {
+  ChatRequestAssistantMessage,
+  ChatRequestMessage,
+  ChatRequestSystemMessage,
+  ChatRequestToolMessage,
+} from "./models.js";
+import { ChatChoiceOutput, ChatCompletionsToolCallOutput } from "./outputModels.js";
+import { isError } from "@azure/core-util";
+import { isUnexpected } from "./isUnexpected.js";
+import { PipelinePolicy } from "@azure/core-rest-pipeline";
+
+const INFERENCE_GEN_AI_SYSTEM_NAME = "az.ai.inference";
+const isContentRecordingEnabled = () =>
+  Boolean(process.env["AZURE_TRACING_GEN_AI_CONTENT_RECORDING_ENABLED"]);
+
+/**
+ * Copied over verbatim from core-tracing
+ */
+function traceAsync<
+  Arguments,
+  ResolvedReturn,
+  PromiseReturn extends Promise<ResolvedReturn> | PromiseLike<ResolvedReturn>,
+>(
+  name: string,
+  args: Arguments,
+  methodToTrace: () => PromiseReturn,
+  onStartTracing?: (span: TracingSpan, args: Arguments) => void,
+  onEndTracing?: (span: TracingSpan, args: Arguments, rt?: ResolvedReturn, error?: unknown) => void,
+  options?: OperationTracingOptions,
+  spanKind?: TracingSpanKind,
+): PromiseReturn {
+  const { span, updatedOptions } = traceClient.startSpan(
+    name,
+    { tracingOptions: options },
+    { spanKind },
+  );
+  options = updatedOptions.tracingOptions;
+  onStartTracing?.(span, args);
+
+  try {
+    return traceClient
+      .withContext(updatedOptions.tracingOptions.tracingContext, methodToTrace)
+      .then(
+        (response) => {
+          onEndTracing?.(span, args, response);
+          return response;
+        },
+        (error) => {
+          onEndTracing?.(span, args, undefined, error);
+          throw error;
+        },
+      ) as PromiseReturn;
+  } catch (err) {
+    onEndTracing?.(span, args, undefined, err);
+    throw err;
+  }
+}
+
+const traceClient = {
+  ...createTracingClient({
+    namespace: "Microsoft.CognitiveServices",
+    packageName: "ai-inference-rest",
+  }),
+  traceAsync,
```

Comment:
Augmenting the core-level client. If you remove the `traceAsync` option you'll be able to just say `const traceClient = createTracingClient({ .... })` etc